go/porcelain/deploy.go

```diff
@@ -404,21 +387,27 @@ func (n *Netlify) uploadFile(ctx context.Context, d *models.Deploy, f *FileBundl
 
 		switch t {
 		case fileUpload:
-			params := operations.NewUploadDeployFileParams().WithDeployID(d.ID).WithPath(f.Name).WithFileBody(f)
-			if timeout != 0 {
-				params.SetTimeout(timeout)
+			body, operationError := os.Open(f.Path)
+			if operationError == nil {
+				defer body.Close()
+				params := operations.NewUploadDeployFileParams().WithDeployID(d.ID).WithPath(f.Name).WithFileBody(body)
+				if timeout != 0 {
+					params.SetTimeout(timeout)
+				}
+				_, operationError = n.Operations.UploadDeployFile(params, authInfo)
 			}
-			_, operationError = n.Operations.UploadDeployFile(params, authInfo)
 		case functionUpload:
 			params := operations.NewUploadDeployFunctionParams().WithDeployID(d.ID).WithName(f.Name).WithFileBody(f)
 			if timeout != 0 {
 				params.SetTimeout(timeout)
 			}
 			_, operationError = n.Operations.UploadDeployFunction(params, authInfo)
+			if operationError != nil {
+				f.Buffer.Seek(0, 0)
+			}
 		}
 
 		if operationError != nil {
-			f.Rewind()
```

Comment:
don't we want to open the file in both cases? Function of file upload? Then rewind in both cases?

Reply:
No. This is exactly what I'm fixing. If there are 10k files, we do not want to open all 10k at once and leave them open. For files (not functions), this PR computes the sums for all the files, and then opens them on demand and they are uploaded.